### PR TITLE
change some references to resource identifiers

### LINKED ISSUE #1853: Change ref to resource
Follow up for backend issue.

### PR DESCRIPTION
**Pull request by acbeni**
# Description

Closes #1853 

# Checklist

- [x] Update release Notes
- [x] Add to the current github milestone 
- [x] Update commercetools api reference

### CODE REVIEW

File: commercetools-models/src/test/java/io/sphere/sdk/carts/commands/CartUpdateCommandIntegrationTest.java [hunk 1]
```diff
@@ -443,7 +443,6 @@ public void setShippingMethod() throws Exception {
                         .isEqualTo(shippingMethod);
                 assertThat(cartWithShippingMethod.getShippingInfo().getShippingMethod().getObj().getTaxCategory().getObj())
                         .as("reference expansion taxCategory")
-                        .isEqualTo(cartWithShippingMethod.getShippingInfo().getTaxCategory().getObj())
```
Comment (yanns) [inline commercetools-models/src/test/java/io/sphere/sdk/carts/commands/CartUpdateCommandIntegrationTest.java]: is that not working anymore?

Comment (acbeni) [reply commercetools-models/src/test/java/io/sphere/sdk/carts/commands/CartUpdateCommandIntegrationTest.java]: Good catch 👍 i tough a `cartWithShippingMethod.getShippingInfo().getTaxCategory()` should have a `ResourceIdentifier` as a return but i was wrong, actually the resource identifier should be only on the draft level

i would fix that.

File: commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraft.java [hunk 2]
```diff
@@ -57,7 +57,7 @@
     Address getShippingAddress();
 
     @Nullable
-    OrderShippingInfo getShippingInfo();
+    ShippingInfoImportDraft getShippingInfo();
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraft.java]: I didn't see this breaking (?) change documented in our release notes?!

File: commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraftImpl.java [hunk 3]
```diff
@@ -157,7 +157,7 @@ public Address getShippingAddress() {
 
     @Override
     @Nullable
-    public OrderShippingInfo getShippingInfo() {
+    public ShippingInfoImportDraft getShippingInfo() {
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraftImpl.java]: I didn't see this mentioned in our release notes?

File: commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraft.java (hunk 2 above)
Comment (acbeni) [reply commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraft.java]: Good catch, would update the doc.

File: commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraftImpl.java (hunk 3 above)
Comment (acbeni) [reply commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraftImpl.java]: actually this class has default visibility so it shouldn't be documented.

**Comment by acbeni** [comment]
@katmatt it's now all clear 👍

**Pull request merged**